src/models/data.ts

```diff
@@ -1,9 +1,9 @@
 // Data is a unit of information about a business
-export interface IData {
+export interface Data {
 	source: string;
 }
 
-// Extracted<Data> contains all extractions from a piece of data of that type
-export interface IExtracted<Data extends IData> {
-	origin: Data;
+// Extracted<DataType> contains all extractions from a piece of data of that type
+export interface Extracted<DataType extends Data> {
```

Comment:
maybe rename it to `ExtractedSource`, since all it will work as is a source of data for the Extraction Provider to work with. That way `ExtractedImage` will extend from `ExtractedSource` and add on extractions